src/components/dock/dock.scss

```diff
@@ -0,0 +1,93 @@
+@use '../../style/mixins';
+@use '../../style/functions';
+@use '../../style/internal/variables';
+
+/**
+* @prop --dock-expanded-max-width: The maximum width of the Dock when it is expanded. Defaults to `max-content` which means the Dock will ajdust its width to the widest dock item.
+* @prop --dock-background-color: Background color of the whole compoentn, defaults to `--contrast-100`.
+* @prop --dock-item-background-color--deselected: Background color of buttons, defaults to `--dock-background-color`.
+* @prop --dock-item-background-color--selected: Background color of selected button, defaults to `--contrast-200`.
+* @prop --dock-item-text-color--deselected: Text of buttons, defaults to `--contrast-1100`.
+* @prop --dock-item-text-color--selected: Text color of selected button, defaults to `--contrast-1300`.
+* @prop --dock-item-icon-color: Color of the optional icons used in each button. Only affects inactive buttons, defaults to text colors for default or selected states.
```

Comment:
is there a need to change all `--dock-item-…` variable names to `--dock-button-…`?

Reply:
Hmm… maybe…

When we supported the option to use a custom component for it, I called the interface `CustomDockButton`. If we support it again in the future, I'd guess that might be the name we'd use. But at the moment we don't support that, so I don't think the consumer will actually see any references to any `dock-button` or `dockButton`. (Unless I'm forgetting something?)

So then it seems like it would make more sense to a consumer that we prefix the css variables with `--dock-item` rather than `--dock-button`. I think? Thoughts? 😅

Reply:
yes you're right. We still send the array as `private dockItems: DockItem[] = [` so it makes sense to call them `--dock-item-`. Then based on what you said regarding future plans, we don't have to change any filename or tag name here, right?
<img width="657" alt="image" src="https://user-images.githubusercontent.com/35954987/180976515-6e30dd71-0645-4893-b40e-a9c30d33d55f.png">
or should we do it for consistency anyways?

Reply:
No, I think `--dock-item` is the better option. And we shouldn't have to change that even if we do allow custom dock buttons in the future. Nothing will really force a custom dock button so use those variables anyway (although a good custom dock button probably should 😄).

Reply:
(I've edited the above. I had written "Nothing will really force a custom dock button **so** use those variables anyway" but meant "Nothing will really force a custom dock button **to** use those variables anyway". That seemed like an important typo to fix 😅)

Reply:
I can tell you that I was contemplating on the comment for a while, and then gave up 😄 

Also didn't really understand if you will rename the files back to original or not. E.g. `dock-button.tsx` -> `dock-item.tsx` . But I guess you won't. Either way I'm fine. :)

Reply:
I hadn't considered renaming the files… 🤔

Anyway, I haven't even gotten started on emitting that event for expanding and collapsing the dock, so I need to get working on that after lunch. All these discussions are distracting me 😅